aspnetcore/tutorials/first-mvc-app/adding-model.md

```diff
@@ -99,278 +108,303 @@ The preceding command adds the [aspnet-codegenerator scaffolding tool](xref:fund
 
 ---
 
-<a name="dc"></a>
-
-## Create a database context class
-
-A database context class is needed to coordinate EF Core functionality (Create, Read, Update, Delete) for the `Movie` model. The database context is derived from [Microsoft.EntityFrameworkCore.DbContext](/dotnet/api/microsoft.entityframeworkcore.dbcontext) and specifies the entities to include in the data model.
-
-Create a *Data* folder.
+Build the project as a check for compiler errors.
 
-Add a *Data/MvcMovieContext.cs* file with the following code: 
+## Scaffold movie pages
 
-[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/zDocOnly/MvcMovieContext.cs?name=snippet)]
+Use the scaffolding tool to produce `Create`, `Read`, `Update`, and `Delete` (CRUD) pages for the movie model.
 
-The preceding code creates a [DbSet\<Movie>](/dotnet/api/microsoft.entityframeworkcore.dbset-1) property for the entity set. In Entity Framework terminology, an entity set typically corresponds to a database table. An entity corresponds to a row in the table.
+# [Visual Studio](#tab/visual-studio)
 
-<a name="reg"></a>
+In **Solution Explorer**, right-click the *Controllers* folder **> Add > New Scaffolded Item**.
 
-## Register the database context
+![view of above step](adding-model/_static/add_controller5.png)
 
-ASP.NET Core is built with [dependency injection (DI)](xref:fundamentals/dependency-injection). Services (such as the EF Core DB context) must be registered with DI during application startup. Components that require these services (such as Razor Pages) are provided these services via constructor parameters. The constructor code that gets a DB context instance is shown later in the tutorial. In this section, you register the database context with the DI container.
+In the **Add Scaffold** dialog, select **MVC Controller with views, using Entity Framework > Add**.
 
-Add the following `using` statements at the top of *Startup.cs*:
+![Add Scaffold dialog](adding-model/_static/add_scaffold5.png)
 
-```csharp
-using MvcMovie.Data;
-using Microsoft.EntityFrameworkCore;
-```
+Complete the **Add MVC Controller with views, using Entity Framework** dialog:
 
-Add the following highlighted code in `Startup.ConfigureServices`:
+* In the **Model class** drop down, select **Movie (MvcMovie.Models)**.
 
-# [Visual Studio](#tab/visual-studio)
+![Add Data context](adding-model/_static/dc5.png)
 
-[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/Startup.cs?name=snippet_ConfigureServices&highlight=5-6)]
+* In the **Data context class** row, select the **+** (plus) sign.
 
-# [Visual Studio Code / Visual Studio for Mac](#tab/visual-studio-code+visual-studio-mac)
+![Add Data context select plus](adding-model/_static/dc5_plus.png)
 
-[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/Startup.cs?name=snippet_UseSqlite&highlight=5-6)]
+* In the **Add Data Context** dialog, the class name *MvcMovie.Data.MvcMovieContext* is generated, select **Add**.
```

Comment:
This is getting way too verbose. Look at https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/model?view=aspnetcore-5.0&tabs=visual-studio#scaffold-the-movie-model and do the same thing. We've had no complaints following that.